password_security/models/res_users.py

```diff
@@ -55,36 +81,23 @@ def _check_password_rules(self, password):
         if not password:
             return True
         company_id = self.company_id
-        message = []
-        if len(list(filter(str.islower, password))
-               ) < company_id.password_lower:
-            message.append('\n ' + _('Lowercase letter (At least ' +
-                                     str(company_id.password_lower) +
-                                     ' character)')
-                           )
-        if len(list(filter(str.isupper, password))
-               ) < company_id.password_upper:
-            message.append('\n ' + 'Uppercase letter (At least ' +
-                           str(company_id.password_upper) + ' character)')
-
-        if len(list(filter(str.isdigit, password))) < \
-                company_id.password_numeric:
-            message.append('\n ' + 'Numeric digit (At least ' +
-                           str(company_id.password_numeric) + ' numeric)')
-
-        if len(re.findall(r"\W", password)) < company_id.password_special:
-            message.append('\n ' + 'Special character (At least ' +
-                           str(company_id.password_special) +
-                           ' special character')
-
-        if len(password) < company_id.password_length:
-            message = ['Password must be %d characters or more.' %
-                       company_id.password_length] + message
-
-        if len(message) > 0:
-            raise PassError('\r'.join(message))
-        else:
-            return True
+        password_regex = ['^']
+        if company_id.password_lower:
```

Comment:
Interestingly enough, I just realized you can outright remove all of these conditions. Now that we're using numerics, an empty option is simply checking for zero instances of the condition:

```
(?=.*?[a-z]){0,}
```